src/payloads/AppUser.ts

```diff
@@ -22,8 +71,13 @@ export class AppUserEventRequest {
   readonly major: string;
 }
 
-export class AppUserEventResponse {
+export class AppUserResponse {
```

Comment:
I see you're overloading AppUserResponse to accomodate roles, any thoughts on having two separate payloads instead? otherwise everything will end up as optional :)

Reply:
My original intention for having most things in AppUserResponse as optional is because of the query parameters from GET '/api/users'. Since we either return an array of everything about the AppUsers or an array of just their first name and last name, I did this so that I won't have to deal with class-validator's way of having more than one type for validateNested (i.e. MultipleEventResponse, MultipleAppUserResponse). I think we'll need inheritance in order to have more than two types for `@Type(() => <type>)`

Reply:
does returning a union type for your route work? then you don't need to make validate nested validate 2 different things

Reply:
Maybe, maybe not. entityToResponse can only convert an entity to one response payload type, so maybe I can have a union type of AppUserResponse and AppUserNameResponse.

Reply:
mm I see, the upside of having two is that we can tag ResponseSchema for the 2 payloads - you'll have more insight since you're working on this directly so ill defer to you